components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImplTest.java

```diff
@@ -0,0 +1,446 @@
+
+        String consumerKey1 = UUID.randomUUID().toString();
+        String authzCodeID1 = UUID.randomUUID().toString();
+        String authzCode1 = UUID.randomUUID().toString();
+        String consumerKey2 = UUID.randomUUID().toString();
+        String authzCodeID2 = UUID.randomUUID().toString();
+        String authzCode2 = UUID.randomUUID().toString();
+        AuthzCodeDO authzCodeDO1 = persistAuthorizationCode(consumerKey1, authzCodeID1, authzCode1, true,
+                OAuthConstants.AuthorizationCodeState.ACTIVE);
+        AuthzCodeDO authzCodeDO2 = persistAuthorizationCode(consumerKey2, authzCodeID2, authzCode2, true,
+                OAuthConstants.AuthorizationCodeState.ACTIVE);
+
+        mockStatic(OAuth2Util.class);
+        mockStatic(IdentityUtil.class);
+        authzCodeDO1.setOauthTokenId(UUID.randomUUID().toString());
+        // If state is EXPIRED/INACTIVE needs to revoke token as well.
+        mockStatic(OAuth2TokenUtil.class);
+        doNothing().when(OAuth2TokenUtil.class, "postRevokeCode", anyString(), anyString(), anyString());
+        List<AuthzCodeDO> revokeAuthzCodeList = new ArrayList<>();
+        revokeAuthzCodeList.add(authzCodeDO1);
+        revokeAuthzCodeList.add(authzCodeDO2);
+        authorizationCodeDAO.deactivateAuthorizationCodes(revokeAuthzCodeList);
+
+        Assert.assertTrue(authorizationCodeDAO.getActiveAuthorizationCodesByConsumerKey
+                (authzCodeDO1.getConsumerKey()).isEmpty());
+        Assert.assertTrue(authorizationCodeDAO.getActiveAuthorizationCodesByConsumerKey
+                (authzCodeDO2.getConsumerKey()).isEmpty());
+    }
+
+    @Test
+    public void testGetLatestAuthorizationCodesByTenant() throws Exception {
+
+        String consumerKey1 = UUID.randomUUID().toString();
+        String authzCodeID1 = UUID.randomUUID().toString();
+        String authzCode1 = UUID.randomUUID().toString();
+        String consumerKey2 = UUID.randomUUID().toString();
+        String authzCodeID2 = UUID.randomUUID().toString();
+        String authzCode2 = UUID.randomUUID().toString();
+        persistAuthorizationCode(consumerKey1, authzCodeID1, authzCode1, true,
+                OAuthConstants.AuthorizationCodeState.ACTIVE);
+        // Creating an application outside default tenant.
+        createApplication(consumerKey2, UUID.randomUUID().toString(), 100);
+        mockStatic(OAuth2Util.class);
+        mockStatic(IdentityUtil.class);
+        when(OAuth2Util.createAuthenticatedUser(anyString(), anyString(), anyString(), anyString())).
+                thenReturn(authenticatedUser);
+
+        Assert.assertFalse(authorizationCodeDAO.getLatestAuthorizationCodesByTenant(DEFAULT_TENANT_ID).isEmpty());
+        Assert.assertTrue(authorizationCodeDAO.getLatestAuthorizationCodesByTenant(100).isEmpty());
+    }
+
+    protected void createApplication(String consumerKey, String consumerSecret, int tenantId) throws Exception {
```

Comment:
why do we need the access modifier as protected?

Reply:
Addressed with bd9a919ce82d35f572e275a65b173e7fdf9e50dc